src/target/riscv/riscv.c

```diff
@@ -5084,189 +5086,140 @@ int riscv_enumerate_triggers(struct target *target)
+		[GDB_REGNO_S1] = { .name = "s1" },
+		[GDB_REGNO_A0] = { .name = "a0" },
+		[GDB_REGNO_A1] = { .name = "a1" },
+		[GDB_REGNO_A2] = { .name = "a2" },
+		[GDB_REGNO_A3] = { .name = "a3" },
+		[GDB_REGNO_A4] = { .name = "a4" },
+		[GDB_REGNO_A5] = { .name = "a5" },
+		[GDB_REGNO_A6] = { .name = "a6" },
+		[GDB_REGNO_A7] = { .name = "a7" },
+		[GDB_REGNO_S2] = { .name = "s2" },
+		[GDB_REGNO_S3] = { .name = "s3" },
+		[GDB_REGNO_S4] = { .name = "s4" },
+		[GDB_REGNO_S5] = { .name = "s5" },
+		[GDB_REGNO_S6] = { .name = "s6" },
+		[GDB_REGNO_S7] = { .name = "s7" },
+		[GDB_REGNO_S8] = { .name = "s8" },
+		[GDB_REGNO_S9] = { .name = "s9" },
+		[GDB_REGNO_S10] = { .name = "s10" },
+		[GDB_REGNO_S11] = { .name = "s11" },
+		[GDB_REGNO_T3] = { .name = "t3" },
+		[GDB_REGNO_T4] = { .name = "t4" },
+		[GDB_REGNO_T5] = { .name = "t5" },
+		[GDB_REGNO_T6] = { .name = "t6" },
+		[GDB_REGNO_PC] = { .name = "pc" },
+		[GDB_REGNO_CSR0] = { .name = "csr0" },
+		[GDB_REGNO_PRIV] = { .name = "priv" },
+		[GDB_REGNO_FT0] = { .name = "ft0" },
+		[GDB_REGNO_FT1] = { .name = "ft1" },
+		[GDB_REGNO_FT2] = { .name = "ft2" },
+		[GDB_REGNO_FT3] = { .name = "ft3" },
+		[GDB_REGNO_FT4] = { .name = "ft4" },
+		[GDB_REGNO_FT5] = { .name = "ft5" },
+		[GDB_REGNO_FT6] = { .name = "ft6" },
+		[GDB_REGNO_FT7] = { .name = "ft7" },
+		[GDB_REGNO_FS0] = { .name = "fs0" },
+		[GDB_REGNO_FS1] = { .name = "fs1" },
+		[GDB_REGNO_FA0] = { .name = "fa0" },
+		[GDB_REGNO_FA1] = { .name = "fa1" },
+		[GDB_REGNO_FA2] = { .name = "fa2" },
+		[GDB_REGNO_FA3] = { .name = "fa3" },
+		[GDB_REGNO_FA4] = { .name = "fa4" },
+		[GDB_REGNO_FA5] = { .name = "fa5" },
+		[GDB_REGNO_FA6] = { .name = "fa6" },
+		[GDB_REGNO_FA7] = { .name = "fa7" },
+		[GDB_REGNO_FS2] = { .name = "fs2" },
+		[GDB_REGNO_FS3] = { .name = "fs3" },
+		[GDB_REGNO_FS4] = { .name = "fs4" },
+		[GDB_REGNO_FS5] = { .name = "fs5" },
+		[GDB_REGNO_FS6] = { .name = "fs6" },
+		[GDB_REGNO_FS7] = { .name = "fs7" },
+		[GDB_REGNO_FS8] = { .name = "fs8" },
+		[GDB_REGNO_FS9] = { .name = "fs9" },
+		[GDB_REGNO_FS10] = { .name = "fs10" },
+		[GDB_REGNO_FS11] = { .name = "fs11" },
+		[GDB_REGNO_FT8] = { .name = "ft8" },
+		[GDB_REGNO_FT9] = { .name = "ft9" },
+		[GDB_REGNO_FT10] = { .name = "ft10" },
+		[GDB_REGNO_FT11] = { .name = "ft11" },
+
+		#define DECLARE_CSR(csr_name, number)[(number) + GDB_REGNO_CSR0] = { .name = #csr_name },
+		#include "encoding.h"
+		#undef DECLARE_CSR
+	};
+
+	static bool initialized;
+
+	if (!initialized) {
+		for (unsigned int i = 0; i < GDB_REGNO_COUNT; ++i) {
+			if (buf[i].name[0])
+				continue;
+			else if (i <= GDB_REGNO_XPR31)
+				initialize_reg_name_with_prefixed_name(&buf[i], "x", i - GDB_REGNO_ZERO);
+			else if (i <= GDB_REGNO_V31 && i >= GDB_REGNO_V0)
+				initialize_reg_name_with_prefixed_name(&buf[i], "v", i - GDB_REGNO_V0);
+			else if (i >= GDB_REGNO_CSR0 && i <= GDB_REGNO_CSR4095)
+				initialize_reg_name_with_prefixed_name(&buf[i], "csr", i - GDB_REGNO_CSR0);
```

Comment:
Replace with the appropriate snprintf(). Then at the end of the for() block check the return value and assert it's less than the max reg name size minus one. (snprintf does not count the null byte in its return value)